File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if i == '(' or i == '[' or i== '{':
                stack.append(i)
            else:
                if not stack:
                    return False
                prev = stack.pop()
                if not((prev == '(' and i == ')') or (prev == '[' and i == ']') or (prev == '{' and i == '}')):
                    return False
        if not stack:
            return True
        return False
                    
                    

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack1 =new Stack<Character>();
        for (char c : s.toCharArray()){
            if (c == '('){
                stack1.push(')');
            }
            else if (c == '['){
                stack1.push(']');
            }
            else if (c == '{'){
                stack1.push('}');
            }
            else if (stack1.isEmpty() || stack1.pop() != c){
                return false;
            }
        }
        if (stack1.isEmpty()){
            return true;
        }
        return false;
    }
}